File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Controllers/ContaController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
using JsonWebToken.Code.Conta;

namespace JsonWebToken.Model
{
  public class ContaController : ApiController
  {
    public IEnumerable<ContaModel> Get()
    {
      ContaBLL bll = new ContaBLL();
      return bll.SelectConta();
    }

    // GET api/<controller>/5
    public string Get(int id)
    {
      // ContaBLL bll = new ContaBLL();
      return "value";
    }

    // POST api/<controller>
    public void Post([FromBody]ContaModel Conta)
    {
      ContaBLL bll = new ContaBLL();

      bll.InsertConta(Conta);
    }

    // PUT api/<controller>/5
    public void Put([FromBody]ContaModel Conta, string acao)
    {
      ContaBLL bll = new ContaBLL();
      if (acao.ToLower().Equals("ativar"))
      {
        bll.AtivarConta(Conta);
      }
      else if (acao.ToLower().Equals("inativar"))
      {
        bll.InativarConta(Conta);
      }

    }

    // DELETE api/<controller>/5
    public void Delete(int id)
    {
      //não realiza esta operacao
      //  ContaBLL bll = new ContaBLL();

    }
  }
}

File: Backend/Administracao.ConciliacaoAPI/Controllers/ContaController.vb
Imports System.Net
Imports System.Net.Http
Imports System.Web.Http
Imports System.Web.Services.Description
Imports Administracao.ConciliacaoAPI.Code.Conta
Imports Administracao.ConciliacaoAPI.Model
Namespace Controllers
    Public Class ContaController
        Inherits ApiController

        ' GET api/<controller>
        Public Function GetValues() As Object
            Try
                Dim bll As New ContaBLL
                Return New ResultModel(bll.SelectConta())
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' GET api/<controller>/5
        Public Function GetValue(ByVal id As Integer) As Object
            Try
                Dim bll As New ContaBLL
                Return New ResultModel(bll.SelectContaPorId(id))
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' POST api/<controller>
        Public Function PostValue(<FromBody()> ByVal ContaModel As ContaModel) As Object
            Try
                Dim bll As New ContaBLL
                bll.InsertConta(ContaModel)
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function

        ' PUT api/<controller>/5
        Public Function PutValue(<FromBody()> ByVal ContaModel As ContaModel, acao As String) As Object
            Try
                Dim bll As New ContaBLL
                bll.UpdateConta(ContaModel, acao)
                Return New ResultModel(Nothing)
            Catch ex As Exception
                Return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message)
            End Try
        End Function
    End Class
End Namespace
